Sources/NumericText/NumericTextModifier.swift

```diff
@@ -21,24 +21,48 @@ public struct NumericTextModifier: ViewModifier {
         _text = text
         _number = number
         self.isDecimalAllowed = isDecimalAllowed
+        
+        
     }
 
     public func body(content: Content) -> some View {
-        content
-            .onChange(of: text) { newValue in
-                let numeric = newValue.numericValue(allowDecimalSeparator: isDecimalAllowed)
-                if newValue != numeric {
-                    text = numeric
-                }
-                number = decimalNumberFormatter.number(from: numeric)
-            }
-            .onChange(of: number, perform: { newValue in
-                if let number = newValue {
-                    text = decimalNumberFormatter.string(from: number) ?? ""
-                } else {
-                    text = ""
-                }
-            })
+        if #available(iOS 14.0, macOS 11, tvOS 14.0, watchOS 7.0, *) {
```

Comment:
What do you think about moving this `#available` check into the `onChangeShimmed` method? It seems a bit wrong that you added the `onChangeShimmed` that works just like the `onChange`, yet have to duplicate the code passed into it here.

Reply:
Sure thing, but this is my first usage of `#available`. Ever.

So I'm open for changes on this one!

Reply:
No worries, it looks pretty good.

Generally I don't mind switching between the two implementations where you are using the thing that you need to availability check. But when you have to make up functions to pass into the method, it seems cleaner to push the availability check a layer down so you aren't duplicating the work, and adding room for subtle differences between the implementations.

Reply:
Done.